dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/AccountController.java

```diff
@@ -178,6 +178,17 @@ public void restoreAccount(
             throw new WebMessageException( WebMessageUtils.conflict( "User does not exist: " + username ) );
         }
 
+        User user = credentials.getUser();
+
+        CredentialsInfo credentialsInfo = new CredentialsInfo( username, password, user.getEmail() != null ? user.getEmail() : "", false );
```

Comment:
Could you just directly use `user.getEmail()`?  Is there some difference whether you provide `null` or an empty string (`""`)? Based on later validation (in UserParameterValidationRule) I think both will fail. Or am I wrong?

Reply:
yes. No need to check for user nullability here as it is validated later on.